openfe_gromacs/protocols/gromacs_md/md_settings.py

```diff
@@ -0,0 +1,668 @@
+
+try:
+    from pydantic.v1 import validator
+except ImportError:
+    from pydantic import validator  # type: ignore[assignment]
+
+
+class SimulationSettings(SettingsBaseModel):
+    """
+    Settings for simulations in Gromacs.
+    """
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    # # # Run control # # #
+    integrator: Literal["md", "sd", "steep"] = "sd"
+    """
+    MD integrators and other algorithms (steep for energy minimization).
+    Allowed values are:
+    'md': a leap-frog integrator
+    'sd': A leap-frog stochastic dynamics integrator. Note that when using this
+          integrator the parameters tcoupl and nsttcouple are ignored.
+    'steep': A steepest descent algorithm for energy minimization.
+    """
+    dt: FloatQuantity["picosecond"] = 0.002 * unit.picosecond
+    """
+    Time step for integration (only makes sense for time-based integrators).
+    Default 0.002 * unit.picosecond
+    """
+    nsteps: int = 0
+    """
+    Maximum number of steps to integrate or minimize, -1 is no maximum
+    Default 0
+    """
+    mass_repartition_factor: int = 1
+    """
+    Scales the masses of the lightest atoms in the system by this factor to
+    the mass mMin. All atoms with a mass lower than mMin also have their mass
+    set to that mMin. Default 1 (no mass scaling)
+    """
+
+    # # # Langevin dynamics # # #
+    ld_seed: int = -1
+    """
+    Integer used to initialize random generator for thermal noise for
+    stochastic and Brownian dynamics. When ld-seed is set to -1,
+    a pseudo random seed is used. Default -1.
+    """
+
+    # # # Neighbor searching # # #
+    cutoff_scheme: Literal["verlet"] = "verlet"
+    """
+    Only allowed option:
+    'verlet': Generate a pair list with buffering. The buffer size is
+              automatically set based on verlet-buffer-tolerance, unless this
+              is set to -1, in which case rlist will be used.
+    """
+    nstlist: int = 10
+    """
+    Frequency to update the neighbor list. When dynamics and
+    verlet-buffer-tolerance set, nstlist is actually a minimum value and
+    gmx mdrun might increase it, unless it is set to 1.
+    If set to zero the neighbor list is only constructed once and never updated.
+    Default 10.
+    """
+    pbc: Literal["xyz", "no", "xy"] = "xyz"
+    """
+    Treatment of periodic boundary conditions.
+    Allowed values are:
+    'xyz': Use periodic boundary conditions in all directions.
+    'no': Use no periodic boundary conditions, ignore the box.
+    'xy': Use periodic boundary conditions in x and y directions only.
+          This can be used in combination with walls.
+    Default 'xyz'.
+    """
+    rlist: FloatQuantity["nanometer"] = 1 * unit.nanometer
+    """
+    Cut-off distance for the short-range neighbor list. With dynamics, this is
+    by default set by the verlet-buffer-tolerance and
+    verlet-buffer-pressure-tolerance options and the value of rlist is ignored.
```

Comment:
Is it worth setting rlist if we don't have these as options?

Reply:
May be useful for energy minimization?